cicd/aggregate-hourly-downloads/main.tf

```diff
@@ -0,0 +1,153 @@
+provider "google" {
+  project = var.gcp_project_id # default inherited by all resources
+  region  = var.gcp_region     # default inherited by all resources
+}
+
+### service account ###
+
+resource "google_service_account" "account" {
+  account_id   = "stats-downloads"
+  display_name = "Service account to deploy aggregate-hourly-downloads cloud function"
+}
+
+resource "google_cloudfunctions2_function_iam_member" "invoker" {
+  cloud_function = google_cloudfunctions2_function.function.name
+  role           = "roles/cloudfunctions.invoker"
+  member         = "serviceAccount:${google_service_account.account.email}"
+}
+
+resource "google_cloud_run_service_iam_member" "cloud_run_invoker" {
+  service = google_cloudfunctions2_function.function.name
+  role    = "roles/run.invoker"
+  member  = "serviceAccount:${google_service_account.account.email}"
+}
+
+resource "google_secret_manager_secret_iam_member" "read_db" {
+  secret_id = var.read_db_pw_secret_name
+  role      = "roles/secretmanager.secretAccessor"
+  member    = "serviceAccount:${google_service_account.account.email}"
+}
+
+resource "google_secret_manager_secret_iam_member" "write_db" {
+  secret_id = var.write_db_pw_secret_name
+  role      = "roles/secretmanager.secretAccessor"
+  member    = "serviceAccount:${google_service_account.account.email}"
+}
+
+resource "google_project_iam_member" "bq_jobs_user" {
+  project = var.gcp_project_id
+  role    = "roles/bigquery.jobUser"
+  member  = "serviceAccount:${google_service_account.account.email}"
+}
+
+resource "google_bigquery_dataset_iam_member" "viewer" {
+  dataset_id = google_bigquery_dataset.dataset.dataset_id
+  role       = "roles/bigquery.dataViewer"
+  member     = "serviceAccount:${google_service_account.account.email}"
+}
+
+resource "google_project_iam_member" "cloudsql_client" {
+  project = var.gcp_project_id
+  role    = "roles/cloudsql.client"
+  member  = "serviceAccount:${google_service_account.account.email}"
+}
+
+### existing bigquery table ###
+
+import {
+  to = google_bigquery_dataset.dataset
+  id = "arxiv_logs"
+}
+
+resource "google_bigquery_dataset" "dataset" {
+  dataset_id = "arxiv_logs"
+
+  lifecycle {
+    prevent_destroy = true # IMPORTANT: block destroy on dataset, use --target to destroy other resources
+  }
+}
+
+### cloud function ###
+
+resource "google_cloudfunctions2_function" "function" {
```

Comment:
Is this a good name "function"? In case we want to combine the code here with a wider terraform project, a more specific name might be good.